tekton/pipeline_run_test.go

```diff
@@ -0,0 +1,137 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tekton
+
+import (
+	"context"
+	"reflect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ExtraParams struct {
+	Name  string
+	Value tektonv1beta1.ArrayOrString
+}
+
+type ReleasePipelineParams struct {
+	Prefix    string
+	Namespace string
+	Extra     ExtraParams
+}
+
+var _ = Describe("PipelineRun", func() {
+
+	var params *ReleasePipelineParams
+	var releasePipelineRun *ReleasePipelineRun
+
+	var release *v1alpha1.Release
+	var releaseParams *v1alpha1.ReleaseSpec
+
+	BeforeEach(func() {
+
+		params = &ReleasePipelineParams{
+			Prefix:    "mypipeline",
+			Namespace: "default",
+			// @todo: use a valid Pipeline params
+			Extra: ExtraParams{
+				Name:  "ExtraParam",
+				Value: tektonv1beta1.ArrayOrString{Type: tektonv1beta1.ParamTypeString, StringVal: "ExtraValue"},
+			},
+		}
+		releaseParams = &v1alpha1.ReleaseSpec{
+			ApplicationSnapshot: "mysnapshot",
+			ReleaseLink:         "myreleaselink",
+		}
+
+		release = &v1alpha1.Release{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "Release",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "myrelease-",
+				Namespace:    params.Namespace,
+				UID:          "8ba3294a-2e1b-4a13-9b3a-faf3cfe3d31b", // This is an arbitrary random UUID
+			},
+			Spec: *releaseParams,
+		}
+
+		ctx := context.Background()
+
+		// The code below sets the ownership for the Release Object
+		kind := reflect.TypeOf(v1alpha1.Release{}).Name()
+		gvk := v1alpha1.GroupVersion.WithKind(kind)
+		controllerRef := metav1.NewControllerRef(release, gvk)
+
+		// creating a release
+		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
+		release.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
+
+		// need to set the Kind as it loses it for reasons...
+		release.TypeMeta.APIVersion = "appstudio.redhat.com/v1alpha1"
+		release.TypeMeta.Kind = "Release"
+
+		// Creates the PipelineRun Object
+		releasePipelineRun = NewReleasePipelineRun(params.Prefix, params.Namespace)
+		Expect(k8sClient.Create(ctx, releasePipelineRun.AsPipelineRun())).Should(Succeed())
+
+	})
+
+	AfterEach(func() {
+		_ = k8sClient.Delete(ctx, release)
+		_ = k8sClient.Delete(ctx, releasePipelineRun.AsPipelineRun())
+	})
+
+	Context("Pipeline Run", func() {
```

Comment:
Can we improve this string a bit? Maybe something like `Create PipelineRun and modify its attributes`

Reply:
Sure. I'll go exactly with this string. I did not have much idea what exactly put there.